### PR TITLE
Feature/canvas lms integration

### PR DESCRIPTION
**Pull request by rajc-aws**
Added feature Canvas LMS integration (an early implementation): 
- using Lex intent/slot matching
- supports resources: announcements, assignments, enrollments, syllabus, and grades
- added CanvasLMSIntegration (with sample questions and course names) as a Examples/Extensions option in Import
- updated readme

Files updated/added:

updated: 
- NOTICE.txt
- templates/examples/extensions/index.js

added: 
- docs/canvaslms_integration.md
- templates/examples/extensions/py_lambda_hooks/CanvasLMSHook/CanvasLMSHelper.py
- templates/examples/extensions/py_lambda_hooks/CanvasLMSHook/CanvasLMSHook.py
- templates/examples/extensions/py_lambda_hooks/CanvasLMSHook/requirements.txt
- templates/examples/extensions/ui_imports/content/CanvasLMSIntegration.json
- templates/examples/extensions/ui_imports/content/CanvasLMSIntegration.txt

Commits for reference: 
- https://github.com/aws-solutions/aws-qnabot/pull/461/commits/5a79962618d477b3e20aa3e95ad9639b5fcc0c54
- https://github.com/aws-solutions/aws-qnabot/pull/461/commits/da9091c86fc3554cea9eec4e26bf9e81f526d258

Note: this branch was created from feature branch {feature/lex-intent-for-qid}.
Only the above mentioned files needs to be merged to {develop}.

Submitting this pull request for review and next steps.

### CODE REVIEW

File: templates/examples/extensions/py_lambda_hooks/CanvasLMSHook/CanvasLMSHelper.py [hunk 1]
```diff
@@ -0,0 +1,419 @@
...
+        set_alt_message (event, return_message)
+
+    return event
+    
+
+"""
+function: query_courses_for_student
+This function retrieves course options across all active enrolled courses, or for a particular course, for the student
+for example: more information about {course name}
+"""
```
Comment (tabdunabi) [inline templates/examples/extensions/py_lambda_hooks/CanvasLMSHook/CanvasLMSHelper.py:131]: please move function documentation inside the function (multiple occurrences).
Example:
```
def query_courses_for_student(event, canvas, student_user_name, course_name_fo_filter):
    """
    function: query_courses_for_student
    This function retrieves course options across all active enrolled courses, or for a particular course, for the student
    for example: more information about {course name}
    """
    ....
```

File: templates/examples/extensions/py_lambda_hooks/CanvasLMSHook/CanvasLMSHelper.py [hunk 2]
```diff
@@ -0,0 +1,419 @@
...
+
+"""
+function: query_enrollments_for_student
+This function retrieves students' active enrollments 
+"""
+def query_enrollments_for_student(event, canvas, student_user_name):
+    # Get the user using user_id to match with LMS SIS_ID
+    user = getCanvasUser (canvas, student_user_name)
+
+    if user:
```
[the remark this answers is not in the document]
Comment (rajc-aws) [reply templates/examples/extensions/py_lambda_hooks/CanvasLMSHook/CanvasLMSHelper.py:103]: Fixed this with exception handling. 
Commit: https://github.com/aws-solutions/aws-qnabot/pull/461/commits/4d8b35beb4301f1a8b7736f8a3eacc8638164882

File: templates/examples/extensions/py_lambda_hooks/CanvasLMSHook/CanvasLMSHelper.py (hunk 1 above)
Comment (rajc-aws) [reply templates/examples/extensions/py_lambda_hooks/CanvasLMSHook/CanvasLMSHelper.py]: Addressed this in latest commit. 
Commit: https://github.com/aws-solutions/aws-qnabot/pull/461/commits/4d8b35beb4301f1a8b7736f8a3eacc8638164882

**Review by tabdunabi — APPROVED** [review]
LGTM.

**Pull request merged**